Service/SchedulerService.php

```diff
@@ -0,0 +1,186 @@
+     *
+     * @var string
+     */
+    private $commandName;
+    
+    /**
+     *
+     * @var ScheduledCommand
+     */
+    private $command;
+    
+    /**
+     * 
+     * @param \Doctrine\Bundle\DoctrineBundle\Registry $doctrine
+     */
+    public function __construct(\Doctrine\Bundle\DoctrineBundle\Registry $doctrine) { $this->doctrine = $doctrine; }
+    
+    /** Aliases */
+    /**
+     * @param $commandName
+     * @return SchedulerService
+     */
+    public function command( $commandName) { return $this->cmd($commandName); }
+
+    /**
+     * @param $commandName
+     * @return SchedulerService
+     */
+    public function get( $commandName) { return $this->cmd($commandName); }
+
+    /**
+     * Set command to handle
+     * 
+     * @param string $commandName
+     * @return SchedulerService
+     */
+    public function cmd( $commandName) {
+        $this->commandName = $commandName;
+        
+        return $this;
+    }
+
+    /**
+     * Check if command exists
+     *
+     * @return bool
+     */
+    public function exists()
+    {
+        try {            
+            if ( $this->getCommand() ) {
+                return true;
+            }
+            return false;
+        } catch (\Exception $ex) {
+            // TODO: Improve error handling            
+            return false;
+        }
+    }
+
+
+    /**
+     * Command Actions
+     */
+    public function run         () { return $this->commandAction('run'); }
+    public function stop        () { return $this->commandAction('stop'); }
+    public function disable     () { return $this->commandAction('disable'); }
+    public function enable      () { return $this->commandAction('enable'); }
+    public function setOndemand () { return $this->commandAction( ScheduledCommand::MODE_ONDEMAND); }
+    public function setAuto     () { return $this->commandAction( ScheduledCommand::MODE_AUTO); }
+    
+    /**
+     * Command Statuses
+     */
+    public function isFailing  () { return $this->commandStatus('failing'); }
+    public function isRunning  () { return $this->commandStatus('running'); }
+    public function isStopped  () { return $this->commandStatus('stopped'); }
+    public function isDisabled () { return $this->commandStatus('disabled'); }
+    public function isEnabled  () { return $this->commandStatus('enabled'); }
+    public function isOndemand () { return $this->commandStatus( ScheduledCommand::MODE_ONDEMAND); }
+    public function istAuto    () { return $this->commandStatus( ScheduledCommand::MODE_AUTO); }
+    
+    /**
+     *
+     * @return ScheduledCommand
+     * @throws \ErrorException
+     */
+    private function getCommand ( ) {
+        if ( $this->command) {
+            return $this->command;
+        }
+        
+        if ( !$this->commandName ){
+            throw new \ErrorException('Missing Command Name.');
+        }
+        
+        $cmd = $this->doctrine->getRepository(ScheduledCommand::class)->findOneBy([
```

Comment:
`$cmd = $this->doctrine->getRepository(ScheduledCommand::class)->findOneBy(['name' => $this>commandName]);`